opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -1886,6 +1886,33 @@ namespace {
     }
 
     template<class T>
+    double
+    FullyImplicitBlackoilSolver<T>::convergenceReduction(const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& B,
+                                                         const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& tempV,
+                                                         const Eigen::Array<double, Eigen::Dynamic, MaxNumPhases>& R,
+                                                         std::array<double,MaxNumPhases>& R_sum,
+                                                         std::array<double,MaxNumPhases>& maxCoeff,
+                                                         std::array<double,MaxNumPhases>& B_avg,
+                                                         int nc) const
+    {
+        const Opm::PhaseUsage& pu = fluid_.phaseUsage();
+        // Do the global reductions
+        for(int idx=0; idx<MaxNumPhases; ++idx)
```

Comment:
There was indeed a problem with the indexing of inactive phases. I
have resorted to use the idx whenever possible.

Additionally I fixed the formatting.